unittests/test_schedulers.py

```diff
@@ -530,6 +530,52 @@ def test_submit_timelimit(self):
         pytest.skip("PBS minimum time limit is 60s")
 
 
+class TestTorqueJob(_TestJob, unittest.TestCase):
+    @property
+    def sched_name(self):
+        return 'torque'
+
+    @property
+    def launcher_name(self):
+        return 'local'
+
+    @property
+    def sched_configured(self):
+        return fixtures.partition_with_scheduler('torque') is not None
+
+    def setup_user(self, msg=None):
+        super().setup_user(msg='Torque not configured')
+
+    def test_prepare(self):
+        self.setup_job()
+        self.testjob.options += ['-l mem=100GB', 'haswell']
+        super().test_prepare()
+        num_nodes = self.testjob.num_tasks // self.testjob.num_tasks_per_node
+        num_cpus_per_node = (self.testjob.num_cpus_per_task *
+                             self.testjob.num_tasks_per_node)
+        expected_directives = set([
+            '#PBS -N "testjob"',
+            '#PBS -l walltime=0:5:0',
+            '#PBS -o %s' % self.testjob.stdout,
+            '#PBS -e %s' % self.testjob.stderr,
+            '#PBS -l nodes=%s:ppn=%s:haswell' % (num_nodes, num_cpus_per_node),
```

Comment:
Since this is the only difference to the PBS scheduler, I'd factor this out as a separate dynamic property and have the test class derive from the `TestPbsJob`.

Reply:
actually this is not the only difference; there's also the testjob options.
i've tried to factor out as much as possible, not sure if this is what you had in mind.

Reply:
It's good overall, I will push some minor changes to that.